internal/mode/static/usage/reporter.go

```diff
@@ -0,0 +1,128 @@
+package usage
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// ClusterDetails are the k8s usage details for the cluster.
+type ClusterDetails struct {
+	// Metadata contains the cluster metadata.
+	Metadata Metadata `json:"metadata"`
+	// PodDetails contain the details about the NGF Pod.
+	PodDetails PodDetails `json:"pod_details"`
+	// NodeCount is the count of Nodes in the cluster.
+	NodeCount int64 `json:"node_count"`
+}
+
+// Metadata contains the cluster metadata.
+type Metadata struct {
+	// DisplayName is a user friendly resource name. It can be used to define
+	// a longer, and less constrained, name for a resource.
+	DisplayName string `json:"displayName"`
+	// UID is the unique identifier for the cluster.
+	UID string `json:"uid"`
+}
+
+// PodDetails contain the details about the NGF Pod.
+type PodDetails struct {
+	// CurrentPodsCount is the total count of NGF NGINX Plus Pods in the cluster.
+	CurrentPodCounts CurrentPodsCount `json:"current_pod_counts"`
+}
+
+// CurrentPodsCount is the total count of NGF NGINX Plus Pods in the cluster.
+type CurrentPodsCount struct {
+	// PodCount is the current count of NGF NGINX Plus Pods in the cluster.
+	PodCount int64 `json:"pod_count"`
+	// DosCount is the count of Pods with NAP DOS enabled in the cluster. Not applicable for NGF,
+	// but required as part of the payload.
+	DosCount int64 `json:"dos_count"`
+	// WafCount is the count of Pods with NAP WAF enabled in the cluster. Not applicable for NGF,
+	// but required as part of the payload.
+	WafCount int64 `json:"waf_count"`
+}
+
+// Reporter reports the NGINX Plus usage info to the provided collector.
+type Reporter struct {
+	// secret contains the credentials for the usage collector.
+	secret SecretStore
+	// endpoint is the API endpoint of the usage collector.
+	endpoint string
+	// insecureSkipVerify controls whether the client verifies the server cert. Used in testing.
+	insecureSkipVerify bool
+}
+
+// NewReporter creates a new usage reporter.
+func NewReporter(
+	secret SecretStore,
+	endpoint string,
+	insecureSkipVerify bool,
+) *Reporter {
+	return &Reporter{
+		secret:             secret,
+		endpoint:           endpoint,
+		insecureSkipVerify: insecureSkipVerify,
+	}
+}
+
+// Report sends a PUT request with the provided data to the API endpoint configured in the Reporter.
+// The clusterUID is used as the name in the API path.
+func (r *Reporter) Report(ctx context.Context, data ClusterDetails) error {
+	buf, err := json.Marshal(data)
+	if err != nil {
```

Comment:
would it make sense to wrap errors in this function adding some additional context?